tests/apollo/util/bft.py

```diff
@@ -630,18 +630,9 @@ def force_quorum_including_replica(self, replica_id, primary=0):
         so that the remaining replicas form a quorum that includes replica_id
         """
         with log.start_action(action_type="force_quorum_including_replica") as action:
-            unstable_replicas = self.all_replicas(without={primary, replica_id})
-
-            random.shuffle(unstable_replicas)
-
-            for backup_replica_id in unstable_replicas:
-                action.log(f'Stopping backup replica {backup_replica_id} in order '
-                      f'to force a quorum including replica {replica_id}...')
-                self.stop_replica(backup_replica_id)
-                if len(self.procs) == 2 * self.config.f + self.config.c + 1:
-                    break
-
-            assert len(self.procs) == 2 * self.config.f + self.config.c + 1
+            assert len(self.procs) >= 2 * self.config.f + self.config.c + 1
+            self.stop_replicas(self.random_set_of_replicas(
+                len(self.procs) - (2 * self.config.f + self.config.c + 1), without={primary, replica_id}))
```

Comment:
Passing the primary as a parameter could lead to inconsistencies. Could you instead dynamically retrieve it, such as:
```python
primary = await bft_network.get_current_primary()
```
and also remove it from the method's parameters?

Reply:
This is going to add some overhead considering currently the method isn't async. Are you sure?

Reply:
Yes

Reply:
PS: don't forget to add `await` when calling the force quorum method